CAN-D/Src/freertos.c

```diff
@@ -9,20 +9,24 @@
 #include "task.h"
 #include "main.h"
 #include "cmsis_os.h"
+#include "can.h"
 
-osThreadId defaultTaskHandle;
+CAN_HandleTypeDef hcan; /* Shared extern from can.h */

```

Comment:
if it is extern, how come its not `extern CAN_HandleTypeDef hcan;`

Reply:
Also though, if we are able to remove all CAN logic from this file, we won't need a `hcan` here.

Reply:
Oops, that can definition shouldn't be there. And we can remove all CAN logic from this file by using a CAN_STARTED flag that tells it if CAN is started.